File: 169-majority-element/majority-element.go
func majorityElement(nums []int) int {
    counts := make(map[int]int)

    for _, num := range nums {
        counts[num]++
    }

    majorityThreshold := len(nums) / 2

    for num, count := range counts {
        if count > majorityThreshold {
            return num
        }
    }

    return -1
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        counts = {}
        
        for num in nums:
            counts[num] = counts.get(num, 0) + 1
            
        majority_threshold = len(nums) // 2
        
        for num, count in counts.items():
            if count > majority_threshold:
                return num
                
        return -1